player/lua/select.lua

```diff
@@ -353,6 +360,101 @@ mp.add_key_binding(nil, "select-audio-device", function ()
     })
 end)
 
+local history_file_path = mp.command_native({"expand-path", options.watch_history_path})
+
+local function save_to_watch_history()
+    local path = mp.command_native({"normalize-path", mp.get_property("path")})
+    local title = mp.get_property("playlist/" .. mp.get_property("playlist-pos") .. "/title")
+
+    local history_file, error_message = io.open(history_file_path, "a")
+    if not history_file then
+        show_error("Failed to write the watch history: " .. error_message)
+        return
+    end
+
+    history_file:write(utils.format_json({os.time(), path, title}) .. "\n")
+    history_file:close()
+end
+
+if options.save_watch_history then
+    mp.register_event("file-loaded", save_to_watch_history)
+end
+
+local function add_history_entry(line, items, paths, osd_playlist_entry)
+    local entry = utils.parse_json(line)
+
+    if not entry then
+        return
+    end
+
+    local time, path, title = unpack(entry)
+
+    local status, date = pcall(os.date, "(%Y-%m-%d %H:%M) ", time)
```

Comment:
Why `pcall`?

Reply:
Because without it one invalid ine in `watch_history.txt` crashes all of select.lua.